packages/@glimmer/application/src/components/component-managers/custom.ts

```diff
@@ -0,0 +1,375 @@
+import { DEBUG } from '@glimmer/env';
+import { consume } from '@glimmer/tracking';
+import { Owner } from '@glimmer/di';
+import { assert } from '@glimmer/util';
+import {
+  ComponentManager as VMComponentManager,
+  ComponentCapabilities,
+  Dict,
+  Option,
+  ProgramSymbolTable,
+  VMArguments,
+  CapturedArguments,
+  JitRuntimeResolver,
+  WithJitStaticLayout,
+  Template,
+  Environment,
+  Invocation,
+  CompilableProgram,
+  Bounds as VMBounds,
+} from '@glimmer/interfaces';
+import { PathReference, Tag, isConst, createTag } from '@glimmer/reference';
+
+import { Capabilities } from '../capabilities';
+import { RootReference } from '../../references';
+import Bounds from '../bounds';
+import { JitComponentDefinition, AotComponentDefinition } from '../component-definition';
+
+export const CAPABILITIES: ComponentCapabilities = {
+  createInstance: true,
+  dynamicLayout: false,
+  dynamicTag: false,
+  wrapped: false,
+  prepareArgs: false,
+  createArgs: true,
+  attributeHook: false,
+  elementHook: false,
+  updateHook: true,
+  createCaller: false,
+  dynamicScope: true,
+};
```

Comment:
FWIW, I think it's silly that Ember uses the same capabilities for _all_ custom components. The point of the design in the RFC is to shim the public capabilities into the underlying glimmer ones. I think we can fix it later, that's not a new thing here, but I never had the chance to mention it.

Reply:
I was wondering about this! Figured we could come back to it later, for sure, but good to know